### PR TITLE
Integrate snippet scanning into FOSSA CLI

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

Integrates snippet scanning into FOSSA CLI as `fossa snippets`.

## Acceptance criteria

Users no longer need to run this as a separate subcommand.

## Testing plan

I tested manually:
```

fossa-cli on ⑆ millhone/integrate [$!] via λ 9.4.7 via 🦀 v1.72.0 took 8s
❯ cabal run fossa snippets analyze -- ~/projects/scratch/src -o ~/projects/scratch/snippets --overwrite-output
 INFO main{target=/Users/jessica/projects/scratch/src/}: Analyzing local snippet matches
 INFO main{target=/Users/jessica/projects/scratch/src/}: Finished matching 20 snippets out of 3 files to 68 matches


fossa-cli on ⑆ millhone/integrate [$!] via λ 9.4.7 via 🦀 v1.72.0
❯ cabal run fossa snippets commit -- ~/projects/scratch/src --analyze-output ~/projects/scratch/snippets --overwrite-fossa-deps
 INFO main{target=/Users/jessica/projects/scratch/src/}: Committing local snippet matches
 INFO main{target=/Users/jessica/projects/scratch/src/}: Wrote output to '/Users/jessica/projects/scratch/src/fossa-deps.yml' with dependency count: 9


fossa-cli on ⑆ millhone/integrate [$!] via λ 9.4.7 via 🦀 v1.72.0
❯ cat /Users/jessica/projects/scratch/src/fossa-deps.yml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /Users/jessica/projects/scratch/src/fossa-deps.yml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ referenced-dependencies:
   2   │ - type: git
   3   │   name: github.com/fs-c/maniac
   4   │   version: v0.7.0-beta
   5   │ - type: git
   6   │   name: github.com/shiquan/PISA
   7   │   version: v0.10a
   8   │ - type: git
   9   │   name: github.com/shiquan/PISA
  10   │   version: v0.10b
  11   │ - type: git
  12   │   name: github.com/shiquan/PISA
  13   │   version: v0.11a
  14   │ - type: git
  15   │   name: github.com/shiquan/PISA
  16   │   version: v0.12
  17   │ - type: git
  18   │   name: github.com/shiquan/PISA
  19   │   version: v0.12a
  20   │ - type: git
  21   │   name: github.com/shiquan/PISA
  22   │   version: v0.4-alpha
  23   │ - type: git
  24   │   name: github.com/shiquan/PISA
  25   │   version: v0.6
  26   │ - type: git
  27   │   name: github.com/wooorm/levenshtein.c
  28   │   version: 2.0.2
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

## Risks

No major risk, although we're taking on some backwards compatibility commitments here.

## Metrics

None

## References

https://fossa.atlassian.net/browse/ANE-1136

## Checklist

- [ ] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [ ] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

### CODE REVIEW

File: docs/references/subcommands/snippets/commit.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## `fossa snippets commit`
+
+This subcommand commits the analysis performed in the `analyze` subcommand into a `fossa-deps` file.
+For more information on possible options, run `millhone commit --help`.
```
Comment (spatten) [inline docs/references/subcommands/snippets/commit.md:4]: Would you have to dump or compile `millhone` to run this command? If so, are there instructions on how to do this that we can link to here?

File: src/App/Fossa/Config/Snippets.hs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    commit = hsubparser . command "commit" $ info commitOpts snippetsCommitInfo
+    commitOpts =
+      CommandCommit
+        <$> baseDirArg
+        <*> switch (long "debug" <> help "Enable debug logging")
+        <*> optional (uriOption (long "endpoint" <> short 'e' <> metavar "URL" <> help "The FOSSA API server base URL (default: https://app.fossa.com)"))
+        <*> optional (strOption (long fossaApiKeyCmdText <> help "the FOSSA API server authentication key (default: FOSSA_API_KEY from env)"))
+        <*> strOption (long "analyze-output" <> help "The directory to which 'analyze' matches were saved")
+        <*> switch (long "overwrite-fossa-deps" <> help "If specified, overwrites the 'fossa-deps' file if it exists")
+        <*> optional (option (eitherReader parseCommitOutputFormat) (long "format" <> help ("The output format for the generated `fossa-deps` file") <> metavar "FORMAT"))
```
Comment (spatten) [inline src/App/Fossa/Config/Snippets.hs:103]: The `--format` parameter is not documented in docs/references/subcommands/snippets/commit.md

File: docs/references/subcommands/snippets/commit.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+## `fossa snippets commit`
+
+This subcommand commits the analysis performed in the `analyze` subcommand into a `fossa-deps` file.
```
Comment (spatten) [inline docs/references/subcommands/snippets/commit.md:3]: Should we link to the [fossa-deps] docs here? https://github.com/fossas/fossa-cli/blob/master/docs/references/files/fossa-deps.md

File: src/App/Fossa/Snippets/Analyze.hs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  , Has Logger sig m
+  ) =>
+  AnalyzeConfig ->
+  m ()
+analyzeWithMillhone conf = withMillhoneBinary $ \bin -> execEffectful root $ mkCmd bin root conf
+  where
+    root = unBaseDir $ analyzeScanDir conf
+
+mkCmd :: BinaryPaths -> Path Abs Dir -> AnalyzeConfig -> Command
+mkCmd bin root AnalyzeConfig{..} =
```
Comment (spatten) [inline src/App/Fossa/Snippets/Analyze.hs:31]: I think it would be worth adding some tests for `mkCmd` here and for Snippets/Commit.hs too

**Review by spatten — APPROVED** [review]
Looks good to me!

I added some minor comments on the docs, and thought we should add a unit test on `mkCmd`, but none of that blocks approval

File: docs/references/subcommands/snippets/commit.md (hunk 1 above)
Comment (jssblck) [reply docs/references/subcommands/snippets/commit.md:4]: Whoops, that was meant to be changed to `fossa snippets`, updated.

File: src/App/Fossa/Snippets/Analyze.hs (hunk 4 above)
Comment (jssblck) [reply src/App/Fossa/Snippets/Analyze.hs:31]: IMO tests for what amounts to argument shuffling don't make a lot of sense (I don't write them for any application, nor for `main` functions, which is effectively what this is).

That aside, we don't actually have a great way to test this today because we'd need to run an actual analysis inside the test to do so. I've filed this ticket to add the ability to test passed arguments: https://fossa.atlassian.net/browse/ANE-1235

File: docs/references/subcommands/snippets/commit.md (hunk 3 above)
Comment (jssblck) [reply docs/references/subcommands/snippets/commit.md:3]: Yep, updated!

File: src/App/Fossa/Snippets/Analyze.hs (hunk 4 above)
Comment (jssblck) [reply src/App/Fossa/Snippets/Analyze.hs:31]: Going to go ahead and merge this for now but happy to talk more on this and do a follow up PR if needed!

Comment (spatten) [reply src/App/Fossa/Snippets/Analyze.hs:31]: Yeah, I'm fine with merging without this. This is on the edge of needing testing and it's not worth doing if it's hard to do.

I think you could do it, though, by exporting `mkCmd` and just testing that. I do that for `themisFlags`, which is pretty much the same thing as `mkCmd`, here: https://github.com/fossas/fossa-cli/blob/master/test/App/Fossa/RunThemisSpec.hs
